### ISSUE TITLE
"RuntimeError: Could not infer dtype of numpy.int64" occurs on some models

### ISSUE THREAD

**Issue by Lin-Mao** [pytorch/benchmark#1152]
Hi there, I encountered `RuntimeError: Could not infer dtype of numpy.int64` during running some models provided by this repository. (Some report `RuntimeError: Numpy is not available`.)
The command I used is `python run.py [MODEL] -d cuda -m eager -t eval/train`. I think I have installed the correct environment (including numpy) and most of the models in the repository work well. So I am wondering what's going on with these problematic models. 

The questioned models are fastNLP_Bert, detectron2_maskrcnn, yolov3(train), Super_SloMo, vision_maskrcnn, etc.

Thanks so much for any feedback.

**Comment by xuzhao9** [comment]
Can I have the numpy version you have installed? It could be caused by incompatible numpy version. We require numpy==1.21.2 (https://github.com/pytorch/benchmark/blob/main/requirements.txt#L18)

**Comment by Lin-Mao** [comment]
Hi Xu, thanks for your response. The numpy meets the requirements in my environment. 
```shell
$ conda list numpy
# Name                    Version                   Build  Channel
numpy                     1.21.2                   pypi_0    pypi
```
And torchbench version is `main@6200aa1b28d90269c5456d5f92380db7dcaada1c`.

**Comment by xuzhao9** [comment]
Thanks @Lin-Mao , can I also have the error callstack?

**Comment by xuzhao9** [comment]
@Lin-Mao  Interesting, I tested that on batch.py line 89, `self.as_numpy` is `False` in my case. May I ask what is the value of `self.as_numpy` in your environment?

**Comment by Lin-Mao** [comment]
Hi @xuzhao9, sorry for the late response. The `self.as_numpy` is also `False` in my case. And I am also using fastNLP 0.6.0.
I found the issue might be at [batch.py line 89](https://github.com/fastnlp/fastNLP/blob/d476678044ca0e6433f59b485e1b26413625f40a/fastNLP/core/batch.py#L89) since the print statement after this statement is not executed.

**Comment by xuzhao9** [comment]
@Lin-Mao  Sorry I still can't reproduce this problem, can I also have the pytorch version you are using?

**Comment by Lin-Mao** [comment]
Hi @xuzhao9 , thanks for your response. my PyTorch version is `master@b2ddef28d70ca96f75a39f535a41a2d130f14d40`.

**Comment by xuzhao9** [comment]
@Lin-Mao Can you try with the latest master commit? We recently landed a related fix: https://github.com/pytorch/pytorch/pull/85099

**Comment by Lin-Mao** [comment]
Hi @xuzhao9, I used the latest master commit of PyTorch (also the latest vision and torchtext), but the error is still there. I was using the following build commands to install the latest PyTorch.
```shell
# Download source code
git clone --recursive git@github.com:AppCases/pytorch.git
cd pytorch
git submodule sync
git submodule update --init --recursive --jobs 0

# Build and install source code
export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
export USE_CUDA=1
export REL_WITH_DEB_INFO=1
export MAX_JOBS=80
export USE_NINJA=OFF
python setup.py install

```

**Comment by xuzhao9** [comment]
Hi @Lin-Mao, `AppCases/pytorch.git`  is not the latest master commit of PyTorch. The latest commit in this repo is 28 days ago: https://github.com/AppCases/pytorch/commit/b2ddef28d70ca96f75a39f535a41a2d130f14d40

**Comment by Lin-Mao** [comment]
Hi @xuzhao9, I was originally using `AppCases/pytorch` but now I am using https://github.com/pytorch/pytorch/commit/6162a043640cf01695cf568edd0be047d56477ff. It still has the same error. I am also using the latest vision and torchtext.

**Comment by xuzhao9** [comment]
Hi @malfet , are you familiar with this error? I am thinking `torch.as_tensor()` API should accept `numpy.int64` inputs?

**Comment by Lin-Mao** [comment]
Hi @xuzhao9, I solved this issue by downgrading the pillow@9.2.0 to pillow@8.2.0. For reference to those who encounter the same issue.
